### DISCUSSION TITLE
Finding gaps and slivers between polygons

### DISCUSSION THREAD

**Discussion by cosmintaran** [NetTopologySuite/NetTopologySuite#478] [Q&A]
Hello.
I'm trying to detect gaps and slivers between polygons, but I cannot do it right. I searched in many places but no succes. 
I have tried to:

    get all intersecting polygons
    cascadeunion with them
    get all polygons from a multipolygon
    check for holes.
    I get a lot of false positive for this case.

Another aproach was to union intersecting polygons in pairs (only two).
The result was a little bit better, but still I get many false positive.

Can anyone please give me a hint on how to do it properly.

Thank you and sorry for my pour English.

**Comment by FObermaier** [comment]
To remove slivers you can use
```
double snapTolerance = 1e-5;
bool clean = true;
var geom = NetTopologySuite.Operation.Overlay.GeometrySnapper.SnapToSelf(geom, snapTolerance, clean);
```

**Reply by cosmintaran** [reply]
Thank you for your answer. It is helpful.
